cla-backend-go/gerrits/service.go

```diff
@@ -71,15 +71,39 @@ func (s service) GetGerrit(gerritID string) (*models.Gerrit, error) {
 }
 
 func (s service) AddGerrit(claGroupID string, projectSFID string, params *models.AddGerritInput) (*models.Gerrit, error) {
-	if params.GroupIDIcla == "" && params.GroupIDCcla == "" {
+	f := logrus.Fields{"functionName": "AddGerrit", "claGroupID": claGroupID, "projectSFID": projectSFID}
+
+	if params.GroupIDIcla == "" && params.GroupIDCcla == "" || params.GroupIDIcla == params.GroupIDCcla {
 		return nil, errors.New("should specify at least a LDAP group for ICLA or CCLA")
 	}
 	if params.GerritName == nil {
 		return nil, errors.New("gerrit_name required")
 	}
+
+	gerritObject, err := s.repo.ExistsByName(*params.GerritName)
+	if err != nil {
+		message := fmt.Sprintf("unable to get gerrit by name : %s", *params.GerritName)
+		log.WithError(err).Warnf(message)
+	}
+
+	if len(gerritObject) > 0 {
+		return nil, errors.New("gerrit_name already present in the system")
+	}
+
 	if params.GerritURL == nil {
 		return nil, errors.New("gerrit_url required")
 	}
+
+	gerritHost, err := extractGerritHost(params.GerritURL.String(), f)
```

Comment:
Actually, if possible - let's clean this all up. We only want to match input that EXACTLY matches the list identified in the swagger spec.  As a result, we can avoid all the http string prefix and host processing and simply do something like: `if params.GerritURL in our list => ok, else fail`

Reply:
I see a bunch of data in DEV that have various paths which are incorrect, such as:

- https://gerrit.linuxfoundation.org/infra/admin/repos/clav2test/icla-and-ccla-project - BAD/INVALID, this will never work as Gerrit doesn't work like this.  You need to simply specify the host.

Should be: "https://gerrit.linuxfoundation.org" ONLY

Reply:
I have skipped the BAD/INVALID host while getting Gerrit instance.